ouroboros-consensus/test-consensus/Test/Consensus/Mempool.hs

```diff
@@ -425,7 +425,10 @@ withTestMempool setup@TestSetup { testLedgerState, testInitialTxs } prop =
       let tracer = Tracer $ \ev -> atomically $ modifyTVar' varEvents (ev:)
 
       -- Open the mempool and add the initial transactions
-      mempool <- openMempoolWithoutSyncThread ledgerInterface cfg tracer
+      mempool <- openMempoolWithoutSyncThread ledgerInterface
+                                              cfg
+                                              (MempoolCapacity 1000) -- TODO @intricate: hardcoded cap
```

Comment:
We can test this by inspecting the trace, e.g.,
* We use a Mempool with capacity 1
* Add a valid transaction
* Add another valid transaction, which should block
* In another thread, we add the first transaction to the ledger
* The first thread resyncs with the updated ledger and removes the first transaction from the Mempool
* Now there's space for our second transaction, and the first thread terminates.
* Finally, inspect the order in which the events were traced, it should be `TraceMempoolAddTxs [tx1] 1`, `TraceMempoolRemoveTxs [tx1] 0`, `TraceMempoolAddTxs [tx2] 1` (with a capacity > 1, the two last ones should be flipped).

Reply:
Implemented this locally and it's already uncovered a few bugs in the implementation. Once I get those all fixed up, will probably add support for testing with capacity > 1.